db/class_test.go

```diff
@@ -0,0 +1,344 @@
+			defer rec.Result().Body.Close()
+		}
+	})
+
+}
+
+func TestGetClass(t *testing.T) {
+	d, err := Open(context.Background(), os.Getenv("TLACFG"))
+	require.NoError(t, err)
+
+	t.Run("Send request to get class", func(t *testing.T) {
+		pr := struct {
+			Uid string
+			Wid string
+		}{
+			userClassOwner.UID,
+			class.WID,
+		}
+		pro, err := json.Marshal(&pr)
+		require.NoError(t, err)
+
+		t.Logf(color_info+"Get class: %s"+color_end, class.WID)
+
+		req, err := http.NewRequest("GET", "/", bytes.NewBuffer(pro))
+		require.NoError(t, err)
+		rec := httptest.NewRecorder()
+		assert.NotNil(t, req, rec)
+
+		req.Header.Set("Content-Type", "application/json")
+		c := echo.New().NewContext(req, rec)
+
+		if assert.NoError(t, d.GetClass(c)) {
+			assert.Equal(t, http.StatusOK, rec.Code)
+			assert.NotEmpty(t, rec.Result().Body)
+			j, err := ioutil.ReadAll(rec.Result().Body)
+			defer rec.Result().Body.Close()
+
+			require.NoError(t, err)
+			assert.NoError(t, json.Unmarshal([]byte(j), &classRet))
+		}
+	})
+
+	t.Run("Check contents of returned class", func(t *testing.T) {
+		assert.Equal(t, class.CID, classRet.CID)
+		assert.Equal(t, class.WID, classRet.WID)
+	})
+}
+
+func TestLeaveClass(t *testing.T) {
+	d, err := Open(context.Background(), os.Getenv("TLACFG"))
+	require.NoError(t, err)
+
+	t.Run("Remove a student from a class", func(t *testing.T) {
+		pr := struct {
+			Uid string
+			Cid string
+		}{
+			user.UID,
+			class.CID,
+		}
+		pro, err := json.Marshal(&pr)
+		require.NoError(t, err)
+
+		t.Logf(color_info+"Leave student: \t%s \nfrom class: \t%s"+color_end, user.UID, class.WID)
+
+		req, err := http.NewRequest("PUT", "/", bytes.NewBuffer(pro))
+		require.NoError(t, err)
+		rec := httptest.NewRecorder()
+		assert.NotNil(t, req, rec)
+
+		req.Header.Set("Content-Type", "application/json")
+		c := echo.New().NewContext(req, rec)
+
+		if assert.NoError(t, d.LeaveClass(c)) {
+			assert.Equal(t, http.StatusOK, rec.Code)
+			assert.NotEmpty(t, rec.Result().Body)
+			defer rec.Result().Body.Close()
+		}
+	})
+
+	t.Run("Get class...", func(t *testing.T) {
+		pr := struct {
+			Uid string
+			Wid string
+		}{
+			userClassOwner.UID,
+			class.WID,
+		}
+		pro, err := json.Marshal(&pr)
+		require.NoError(t, err)
+
+		req, err := http.NewRequest("GET", "/", bytes.NewBuffer(pro))
+		require.NoError(t, err)
+		rec := httptest.NewRecorder()
+		assert.NotNil(t, req, rec)
+
+		req.Header.Set("Content-Type", "application/json")
+		c := echo.New().NewContext(req, rec)
+
+		if assert.NoError(t, d.GetClass(c)) {
+			assert.Equal(t, http.StatusOK, rec.Code)
+			assert.NotEmpty(t, rec.Result().Body)
+			j, err := ioutil.ReadAll(rec.Result().Body)
+			defer rec.Result().Body.Close()
+
+			require.NoError(t, err)
+			assert.NoError(t, json.Unmarshal([]byte(j), &classRet))
+			assert.Equal(t, class.CID, classRet.CID)
+		}
+	})
+
+	t.Run("... and check its contents", func(t *testing.T) {
+		assert.Empty(t, classRet.Members)
+		// TODO: more testing
+	})
```

Comment:
Bear in mind that each call of `Testing.T.Run()` will spawn a new goroutine, so this test case might be poorly defined.

Reply:
Does that mean test cases for class should be a single, large test function since it relies on sequential execution a lot?